src/createServer.js

```diff
@@ -2,7 +1,57 @@
+const http = require('http');
+const url = require('url');
+const fs = require('fs');
 
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  const server = http.createServer((req, res) => {
+    const normalizedUrl = new url.URL(req.url, `http://${req.headers.host}`);
+
+    const pathname = normalizedUrl.pathname.slice(1) || 'index.html';
+
+    res.setHeader('Content-Type', 'text/plain');
+
+    if (!req.url.startsWith('/file')) {
+      res.statusCode = 400;
+
+      res.end();
+
+      return;
+    }
+
+    if (!req.url.startsWith('/file/')) {
+      res.end('hint message');
+
+      return;
+    }
+
+    if (req.url.includes('//')) {
+      res.statusCode = 404;
+
+      res.end();
+
+      return;
+    }
+
+    const path = pathname.slice(5);
+
+    if (!fs.existsSync(`public/${path}`)) {
+      res.statusCode = 404;
+      res.end('File not found');
+
+      return;
+    }
+
+    fs.readFile(`public/${path}`, (err, data) => {
+      if (!err) {
+        res.end(data);
+      }
+
+      res.statusCode = 404;
+      res.end();
```

Comment:
The error handling in the `fs.readFile` callback is incorrect. Currently, if there is no error, the response is ended with the file data, but if there is an error, the response is ended with a 404 status code without checking the error. You should check for the error and handle it appropriately, possibly logging the error or sending a more descriptive error message.